.github/workflows/pull.yml

```diff
@@ -199,9 +199,6 @@ jobs:
 
         PYTHON_EXECUTABLE=python bash .ci/scripts/setup-linux.sh "cmake"
 
-        # install pybind
```

Comment:
I agree that we should stop double-installing, but I'd much rather remove the bespoke CI-only installation stuff in favor of install_requirements. I think that if we remove these calls to install_requirements, then none (or very few) of the CI jobs will actually exercise install_requirements. And removing the CI-only logic would reduce overall maintenance costs.